src/kademlia/node.cpp

```diff
@@ -317,7 +317,8 @@ void node::incoming(msg const& m)
 			// responds to 'query' messages that it receives.
 			if (m_settings.read_only) break;
 
-			if (!native_address(m.addr)) break;
+			// only respond to requests if they're addressed to this node
```

Comment:
is there a good reason to only filter requests (``q``) and not responses (``r``) based on which socket we received it on?

Reply:
Nodes can send requests using a listen socket other than their own for cross-network bootstrap. They need to be able to receive the response.